sys/kern/vm_map.c

```diff
@@ -237,8 +205,8 @@ static int vm_map_findspace_nolock(vm_map_t *map, vaddr_t /*inout*/ *start_p,
   assert(page_aligned_p(start) && page_aligned_p(length));
 
   /* Bounds check */
-  start = max(start, vm_map_start(map));
-  if (start + length > vm_map_end(map))
+  start = max(start, (vaddr_t)USER_SPACE_BEGIN);
```

Comment:
If gcc is annoying here you can change definition of `USER_SPACE_BEGIN` to `((vaddr_t) 0xmagic)` since it's always used in `vaddr_t` context.